File: src/main/java/at/yawk/math/data/MultiplicationExpression.java
package at.yawk.math.data;

/**
 * @author yawkat
 */
public class MultiplicationExpression extends BinaryExpression {
    protected MultiplicationExpression(Expression left, Expression right) {
        super(left, right);
    }

    @Override
    protected String toString(String lhs, String rhs) {
        return '(' + lhs + ") * (" + rhs + ')';
    }
}

File: src/main/kotlin/at/yawk/math/data/MultiplicationExpression.kt
package at.yawk.math.data

import at.yawk.math.EqualsHelper

/**
 * @author yawkat
 */
open class MultiplicationExpression(components: List<Expression>) : ChainExpression(components) {
    override fun withComponents(components: List<Expression>): ChainExpression {
        return MultiplicationExpression(components)
    }

    override fun toString(radix: Int): String {
        return components.map { it.toString(radix) }.joinToString(" * ")
    }

    override fun equals(other: Any?): Boolean {
        return EqualsHelper.equals<MultiplicationExpression>(other, { it.components == components })
    }

    override fun hashCode(): Int {
        return EqualsHelper.hashCode(components)
    }
}